### PR TITLE
Feature/testing

### PR DESCRIPTION
**Pull request by Roman-Robert**

### CODE REVIEW

File: src/main/java/sortingApp/Input.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+public class Input {
+    /**
+     * The inputArray method accepts integer values from console.
+     * The method accepts no more than 10 integer values (or up to the first not-integer value received).
+     * @return As a result, method returns ArrayList with entered integer values (in order of input).
+     */
+    public static ArrayList<Integer> inputArray() {
+        Scanner scanner = new Scanner(System.in);
+        ArrayList<Integer> arrayList = new ArrayList<>();
```
Comment (OlgaPoddubnaia) [inline src/main/java/sortingApp/Input.java:14]: after line 14 there should be empty line according to code convention

File: src/test/java/EmptyCaseTest.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import org.junit.Assert;
+import org.junit.Test;
+import sortingApp.Sorting;
+
+import java.util.ArrayList;
+
+public class EmptyCaseTest extends Sorting {
+    @Test
+    public void testZeroCase() {
+        ArrayList<Integer> emptyArrayList = new ArrayList<>();
```
Comment (OlgaPoddubnaia) [inline src/test/java/EmptyCaseTest.java:10]: after line 10 there should be empty line according to code convention

File: src/test/java/MoreThatTenElementCaseTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.ArrayList;
+
+@RunWith(Parameterized.class)
+public class MoreThatTenElementCaseTest extends Sorting {
+
+    @Parameterized.Parameters
+    public static ArrayList<ArrayList<Integer>> groupOfArrays() {
+        ArrayList<ArrayList<Integer>> arrayList = new ArrayList<>();
+        ArrayList<Integer> arrayListOne = new ArrayList<>();
+        ArrayList<Integer> arrayListTwo = new ArrayList<>();
```
Comment (OlgaPoddubnaia) [inline src/test/java/MoreThatTenElementCaseTest.java:15]: after line 15 there should be empty line according to code convention
"Between the local variables in a method and its first statement"

File: .idea/encodings.xml [hunk 4]
```diff
@@ -2,5 +2,6 @@
 <project version="4">
   <component name="Encoding">
     <file url="file://$PROJECT_DIR$/src/main/java" charset="UTF-8" />
+    <file url="file://$PROJECT_DIR$/src/main/resources" charset="UTF-8" />
```
Comment (OlgaPoddubnaia) [inline .idea/encodings.xml:5]: Please add ".idea/" into file "sorting-app\.gitignore". 
for files to be deleted from remote repository you can delete them, commit and then restore with ctrl+z

Comment (OlgaPoddubnaia) [reply .idea/encodings.xml:5]: filed under .idea folder should not be pushed into remote repository

File: src/main/java/sortingApp/Input.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package sortingApp;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Input {
+    /**
+     * The inputArray method accepts integer values from console.
```
Comment (OlgaPoddubnaia) [inline src/main/java/sortingApp/Input.java:8]: "The inputArray" could be deleted

File: src/main/java/sortingApp/Input.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+     * The method accepts no more than 10 integer values (or up to the first not-integer value received).
+     * @return As a result, method returns ArrayList with entered integer values (in order of input).
```
Comment (OlgaPoddubnaia) [inline src/main/java/sortingApp/Input.java:10]: "As a result, method returns" could be del

File: src/main/java/sortingApp/Input.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
+package sortingApp;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Input {
```
Comment (OlgaPoddubnaia) [inline src/main/java/sortingApp/Input.java:6]: add an empty line before javadoc

File: src/main/java/sortingApp/Sorting.java [hunk 8]
```diff
@@ -0,0 +1,19 @@
+package sortingApp;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class Sorting {
```
Comment (OlgaPoddubnaia) [inline src/main/java/sortingApp/Sorting.java:6]: add an empty line before javadoc

File: src/main/java/sortingApp/Input.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    /**
+     * The inputArray method accepts integer values from console.
+     * The method accepts no more than 10 integer values (or up to the first not-integer value received).
```
Comment (OlgaPoddubnaia) [inline src/main/java/sortingApp/Input.java:9]: there should be empty line before @tags accroding to this [https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#principles](url)  "Insert a blank comment line between the description and the list of tags, as shown."

File: src/main/java/sortingApp/Sorting.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    /**
+     * The sortingArray method takes ArrayList with integer values and sorts it in ascending order.
```
Comment (OlgaPoddubnaia) [inline src/main/java/sortingApp/Sorting.java:8]: there should be empty line before @tags according to this [https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#principles](url)   "Insert a blank comment line between the description and the list of tags, as shown."

File: pom.xml [hunk 11]
```diff
@@ -14,20 +14,36 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     </properties>
```
Comment (OlgaPoddubnaia) [inline pom.xml:16]: please, do ctrl+alt+L for this file to format content

File: pom.xml [hunk 12]
```diff
@@ -14,20 +14,36 @@
...
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-jar-plugin</artifactId>
+            <version>3.3.0</version>
+        </dependency>
     </dependencies>
-</project>
\ No newline at end of file
+
+
+
```
Comment (OlgaPoddubnaia) [inline pom.xml:48]: del empty lines 46-48

File: src/log4j.properties [hunk 13]
```diff
@@ -0,0 +1,9 @@
+# Set root logger level to DEBUG and its only appender to A1.
+log4j.rootLogger=INFO, A1
+
```
Comment (OlgaPoddubnaia) [inline src/log4j.properties:3]: for a file src/log4j.properties create a folder "resources" under src folder

File: src/test/java/NullCaseTest.java [hunk 14]
```diff
@@ -0,0 +1,10 @@
+import org.junit.Assert;
+import org.junit.Test;
+import sortingApp.Sorting;
+
+public class NullCaseTest extends Sorting {
+    @Test(expected = NullPointerException.class)
```
Comment (OlgaPoddubnaia) [inline src/test/java/NullCaseTest.java:6]: for all test please write a javadoc

File: src/test/java/EmptyCaseTest.java [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+        Assert.assertEquals(sortingArray(emptyArrayList), emptyArrayList);
```
Comment (OlgaPoddubnaia) [inline src/test/java/EmptyCaseTest.java:11]: in my opinion, the app shouldn't work with empty arraylist

**Comment by OlgaPoddubnaia** [comment]
please, add test where it will be checked if a String was entered, double

File: src/main/java/sortingApp/Input.java [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+     * @return As a result, method returns ArrayList with entered integer values (in order of input).
+     */
+    public static ArrayList<Integer> inputArray() {
+        Scanner scanner = new Scanner(System.in);
+        ArrayList<Integer> arrayList = new ArrayList<>();
+        System.out.println("Enter a sequence of integers values");
+        for (int i = 0; i < 10; i++) {
+            if (scanner.hasNextInt()) arrayList.add(scanner.nextInt());
+            else {
+                System.out.println("Result:");
```
Comment (OlgaPoddubnaia) [inline src/main/java/sortingApp/Input.java:19]: instead of System.out.println("Result:"); it should be something like "Non-integer value was entered. Input is over"

**Review by OlgaPoddubnaia — COMMENTED** [review]
please, make changes according to comments

File: src/main/java/sortingApp/Input.java (hunk 1 above)
Comment (Roman-Robert) [reply src/main/java/sortingApp/Input.java:14]: done

File: src/test/java/EmptyCaseTest.java (hunk 2 above)
Comment (Roman-Robert) [reply src/test/java/EmptyCaseTest.java:10]: done

File: src/test/java/MoreThatTenElementCaseTest.java (hunk 3 above)
Comment (Roman-Robert) [reply src/test/java/MoreThatTenElementCaseTest.java:15]: +

File: .idea/encodings.xml (hunk 4 above)
Comment (Roman-Robert) [reply .idea/encodings.xml:5]: complete

File: src/main/java/sortingApp/Input.java (hunk 5 above)
Comment (Roman-Robert) [reply src/main/java/sortingApp/Input.java:8]: ok

File: src/main/java/sortingApp/Input.java (hunk 6 above)
Comment (Roman-Robert) [reply src/main/java/sortingApp/Input.java:10]: ok

File: src/main/java/sortingApp/Input.java (hunk 7 above)
Comment (Roman-Robert) [reply src/main/java/sortingApp/Input.java:6]: ok

File: src/main/java/sortingApp/Sorting.java (hunk 8 above)
Comment (Roman-Robert) [reply src/main/java/sortingApp/Sorting.java:6]: ok

File: src/main/java/sortingApp/Input.java (hunk 9 above)
Comment (Roman-Robert) [reply src/main/java/sortingApp/Input.java:9]: ok

File: src/main/java/sortingApp/Sorting.java (hunk 10 above)
Comment (Roman-Robert) [reply src/main/java/sortingApp/Sorting.java:8]: ok

File: pom.xml (hunk 11 above)
Comment (Roman-Robert) [reply pom.xml:16]: I did it, already

File: pom.xml (hunk 12 above)
Comment (Roman-Robert) [reply pom.xml:48]: ok

File: src/log4j.properties (hunk 13 above)
Comment (Roman-Robert) [reply src/log4j.properties:3]: done

File: src/test/java/EmptyCaseTest.java (hunk 15 above)
Comment (Roman-Robert) [reply src/test/java/EmptyCaseTest.java:11]: I'm disagree with you. I will ask mentor about this corner condition.

File: src/main/java/sortingApp/Input.java (hunk 16 above)
Comment (Roman-Robert) [reply src/main/java/sortingApp/Input.java:19]: done
